Assets/_BForBoss/_Core/Scripts/WaveSystem/WaveModel.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+
+namespace BForBoss
+{
+    public class WaveModel
```

Comment:
This is how i imagined this is implemented.  
Only expose
1. functions that increment/decrement numbers by 1. 
2. event Action that allow subscription.

And that's it. 
```
public class WaveModel {
    private int _killCount = 0;

    private int KillCount {
         get => _killCount;
         set => {
              _killCount = value;
              OnKillCountChanged?.Invoke(value)
         }
    }

    public int IncrementKillCount(){
          KillCount++;
    }

    public event Action<Int> OnKillCountChanged;
}
```